File: src/heap_type.ml
type t

external ext : unit -> t = "caml_binaryen_heap_type_ext"
external func : unit -> t = "caml_binaryen_heap_type_func"
external any : unit -> t = "caml_binaryen_heap_type_any"
external eq : unit -> t = "caml_binaryen_heap_type_eq"
external i31 : unit -> t = "caml_binaryen_heap_type_i31"
external struct_ : unit -> t = "caml_binaryen_heap_type_struct"
external array : unit -> t = "caml_binaryen_heap_type_array"
external string : unit -> t = "caml_binaryen_heap_type_string"
external none : unit -> t = "caml_binaryen_heap_type_none"
external noext : unit -> t = "caml_binaryen_heap_type_noext"
external nofunc : unit -> t = "caml_binaryen_heap_type_nofunc"
external is_basic : t -> bool = "caml_binaryen_heap_type_is_basic"
external is_signature : t -> bool = "caml_binaryen_heap_type_is_signature"
external is_struct : t -> bool = "caml_binaryen_heap_type_is_struct"
external is_array : t -> bool = "caml_binaryen_heap_type_is_array"
external is_bottom : t -> bool = "caml_binaryen_heap_type_is_bottom"
external get_bottom : t -> t = "caml_binaryen_heap_type_get_bottom"
external is_sub_type : t -> bool = "caml_binaryen_heap_type_is_sub_type"

external set_type_name : Module.t -> t -> string -> unit
  = "caml_binaryen_module_set_type_name"
(** Sets the textual name of a compound `heapType`. Has no effect if the type
    already has a canonical name. *)

external set_field_name : Module.t -> t -> int -> string -> unit
  = "caml_binaryen_module_set_field_name"
(** Sets the field name of a struct `heapType` at index `index`. *)

File: src/heap_type.js
//Provides: caml_binaryen_heap_type_ext
//Requires: Binaryen
function caml_binaryen_heap_type_ext() {
  return Binaryen.HeapType.extern;
}

//Provides: caml_binaryen_heap_type_func
//Requires: Binaryen
function caml_binaryen_heap_type_func() {
  return Binaryen.HeapType.func;
}

//Provides: caml_binaryen_heap_type_any
//Requires: Binaryen
function caml_binaryen_heap_type_any() {
  return Binaryen.HeapType.any;
}

//Provides: caml_binaryen_heap_type_eq
//Requires: Binaryen
function caml_binaryen_heap_type_eq() {
  return Binaryen.HeapType.eq;
}

//Provides: caml_binaryen_heap_type_i31
//Requires: Binaryen
function caml_binaryen_heap_type_i31() {
  return Binaryen.HeapType.i31;
}

//Provides: caml_binaryen_heap_type_struct
//Requires: Binaryen
function caml_binaryen_heap_type_struct() {
  return Binaryen.HeapType.struct;
}

//Provides: caml_binaryen_heap_type_array
//Requires: Binaryen
function caml_binaryen_heap_type_array() {
  return Binaryen.HeapType.array;
}

//Provides: caml_binaryen_heap_type_string
//Requires: Binaryen
function caml_binaryen_heap_type_string() {
  return Binaryen.HeapType.string;
}

//Provides: caml_binaryen_heap_type_none
//Requires: Binaryen
function caml_binaryen_heap_type_none() {
  return Binaryen.HeapType.none;
}

//Provides: caml_binaryen_heap_type_noext
//Requires: Binaryen
function caml_binaryen_heap_type_noext() {
  return Binaryen.HeapType.noextern;
}

//Provides: caml_binaryen_heap_type_nofunc
//Requires: Binaryen
function caml_binaryen_heap_type_nofunc() {
  return Binaryen.HeapType.nofunc;
}

//Provides: caml_binaryen_heap_type_exn
//Requires: Binaryen
function caml_binaryen_heap_type_exn() {
  return Binaryen.HeapType.exn;
}

//Provides: caml_binaryen_heap_type_noexn
//Requires: Binaryen
function caml_binaryen_heap_type_noexn() {
  return Binaryen.HeapType.noexn;
}

//Provides: caml_binaryen_heap_type_is_basic
//Requires: Binaryen
function caml_binaryen_heap_type_is_basic(ty) {
  return Binaryen._BinaryenHeapTypeIsBasic(ty);
}

//Provides: caml_binaryen_heap_type_is_signature
//Requires: Binaryen
function caml_binaryen_heap_type_is_signature(ty) {
  return Binaryen._BinaryenHeapTypeIsSignature(ty);
}

//Provides: caml_binaryen_heap_type_is_struct
//Requires: Binaryen
function caml_binaryen_heap_type_is_struct(ty) {
  return Binaryen._BinaryenHeapTypeIsStruct(ty);
}

//Provides: caml_binaryen_heap_type_is_array
//Requires: Binaryen
function caml_binaryen_heap_type_is_array(ty) {
  return Binaryen._BinaryenHeapTypeIsArray(ty);
}

//Provides: caml_binaryen_heap_type_is_bottom
//Requires: Binaryen
function caml_binaryen_heap_type_is_bottom(ty) {
  return Binaryen._BinaryenHeapTypeIsBottom(ty);
}

//Provides: caml_binaryen_heap_type_get_bottom
//Requires: Binaryen
function caml_binaryen_heap_type_get_bottom(ty) {
  return Binaryen._BinaryenHeapTypeGetBottom(ty);
}

//Provides: caml_binaryen_heap_type_is_sub_type
//Requires: Binaryen
function caml_binaryen_heap_type_is_sub_type(left, right) {
  return Binaryen._BinaryenHeapTypeIsSubType(left, right);
}

//Provides: caml_binaryen_type_get_heap_type
//Requires: Binaryen
function caml_binaryen_type_get_heap_type(typ) {
  return Binaryen._BinaryenTypeGetHeapType(typ);
}

//Provides: caml_binaryen_module_set_type_name
//Requires: caml_jsstring_of_string
function caml_binaryen_module_set_type_name(wasm_mod, heapType, name) {
  wasm_mod.setTypeName(heapType, caml_jsstring_of_string(name));
}

//Provides: caml_binaryen_module_set_field_name
//Requires: caml_jsstring_of_string
function caml_binaryen_module_set_field_name(wasm_mod, heapType, index, name) {
  wasm_mod.setFieldName(heapType, index, caml_jsstring_of_string(name));
}